### PR TITLE
Introduce Kryo Serializer to Streamlet

### PR DESCRIPTION
**Pull request by srkukarni**
Currently Streamlet relies only on Java serialization which has very bad performance. This pr allows topologies to use Kryo serialization.

### CODE REVIEW

**Comment by srkukarni** [comment]
@jerrypeng

File: heron/api/src/java/com/twitter/heron/streamlet/KeyValue.java [hunk 1]
```diff
@@ -31,6 +31,10 @@
     return new KeyValue<R, T>(k, v);
   }
 
+  public KeyValue() {
+    // nothing really
```
Comment (nwangtw) [inline heron/api/src/java/com/twitter/heron/streamlet/KeyValue.java]: what is this dummy ctor for? Is it better to have mis-usage protection if this is required by kryo?

File: heron/api/src/java/com/twitter/heron/streamlet/Config.java [hunk 2]
```diff
@@ -44,7 +45,7 @@ public static Config defaultConfig() {
         .build();
   }
 
-  com.twitter.heron.api.Config getHeronConfig() {
+  public com.twitter.heron.api.Config getHeronConfig() {
```
Comment (nwangtw) [inline heron/api/src/java/com/twitter/heron/streamlet/Config.java]: Is this change necessary?

File: heron/api/src/java/com/twitter/heron/streamlet/Config.java [hunk 3]
```diff
@@ -107,6 +108,19 @@ public Builder setUserConfig(String key, Object value) {
...
 
+    /**
+     * Sets the topology to use the Kryo serializer for serializing
+     * streamlet elements
+     */
+    public Builder useKryoSerializer() {
+      try {
+        config.setSerializationClassName(new KryoSerializer().getClass().getName());
+      } catch (NoClassDefFoundError e) {
+        throw new RuntimeException("Please link with kryo");
```
Comment (nwangtw) [inline heron/api/src/java/com/twitter/heron/streamlet/Config.java]: Suggest to be more explicit in error message, like "Linking with kryo is needed because useKryoSerializer is used".

File: heron/api/src/java/com/twitter/heron/streamlet/Config.java (hunk 2 above)
Comment (srkukarni) [reply heron/api/src/java/com/twitter/heron/streamlet/Config.java]: Not needed. Reverted

File: heron/api/src/java/com/twitter/heron/streamlet/Config.java (hunk 3 above)
Comment (srkukarni) [reply heron/api/src/java/com/twitter/heron/streamlet/Config.java]: Changed

File: heron/api/src/java/com/twitter/heron/streamlet/KeyValue.java (hunk 1 above)
Comment (srkukarni) [reply heron/api/src/java/com/twitter/heron/streamlet/KeyValue.java]: This is primarily for Kryo. Made it non public
